userspace/libsinsp/plugin.cpp

```diff
@@ -80,6 +80,12 @@ const char* sinsp_plugin::get_owner_last_error(ss_plugin_owner_t* o)
 	return t->m_last_owner_err.c_str();
 }
 
+static void plugin_log_fn(const char* component, const char* msg, ss_plugin_log_severity sev)
+{
+	std::string prefix = (component == NULL) ? "" : std::string(component) + ": ";
```

Comment:
What if, when component is null, we fallback at plugin name?
If user then wants to set it empty, they need to pass an empty string (non-null).

Reply:
I'm not sure that we can access the plugin name from a static function.
To do that we should make `plugin_log_fn` a method of `sinsp_plugin`. WDYT?

Reply:
ah i see, you are right. cc @jasondellaluce wdyt?

Reply:
Good point Fede. That would require passing the plugin instance through an opaque pointer though, effectively adding one extra parameters to the function signature. I think it makes sense, I just tried to avoid it to make the UX less cumbersome.

Reply:
I agree! This will make the API more complex user-side, but I think it's worth it.
Especially if we plan to ditch the global logger, in that case we will need to access the plugin instance anyway.